sql/sql_parse.cc

```diff
@@ -7992,11 +7992,22 @@ static bool wsrep_should_retry_in_autocommit(enum_sql_command &sql_command)
     same symptom is found for other commands, then please add it to the
     below list.
   */
+
+  enum_sql_command sql_command= thd->lex->sql_command;
   switch (sql_command)
   {
     case SQLCOM_CHECK:
     case SQLCOM_SELECT:
       return false;
+    case SQLCOM_ALTER_TABLE:
+    {
+      if (thd->lex->alter_info.flags & Alter_info::ALTER_ADMIN_PARTITION)
+      {
+        return false;
+      }
+      else
+        return true;
+    }
```

Comment:
CHECK TABLE doesn't return rows to the client, right?

mysql> check table t1;
+---------+-------+----------+----------+
| Table   | Op    | Msg_type | Msg_text |
+---------+-------+----------+----------+
| test.t1 | check | status   | OK       |
+---------+-------+----------+----------+

why is CHECK TABLE /ALTER CHECK special?

Also, instead of aborting retry, cant we reset (or end the current data sent to client) and retry transaction?

Reply:
>CHECK TABLE doesn't return rows to the client, right?

It does send a row and the row itself is
```
+---------+-------+----------+----------+
| Table | Op | Msg_type | Msg_text |
+---------+-------+----------+----------+
| test.t1 | check | status | OK |
+---------+-------+----------+----------+
```
> why is CHECK TABLE /ALTER CHECK special?

This is because CHECK TABLE is not replicated and any High Priority Transaction can bf-abort CHECK TABLE/ALTER CHECK command.


>Also, instead of aborting retry, cant we reset (or end the current data sent to client) and retry transaction?

I checked the implementation of protocol_classic, and I didn't see any functions for clearing the net buffers. May be it was by design that there should not be a way to clear them (probably to catch bugs like this). I'm not sure.

IMO, implementing it from scratch may have unforeseen effects on protocol/client (we will never know), and may also need lot of testing. So, let us leave it as is for now.

Reply:
ok, this is considered as single row. Instead of relying on thd->lex->sql_command, is there any other way of knowing we have sent at least a single row? this would be generic

Reply:
I see `Protocol_classic::get_output_pkt_nr()` function to check the number of packets written to net buffers, but this function is not used anywhere in the code to check our assumption is valid or not.

Reply:
Also, since `net.pkt_nr` is incremented on every `my_net_write()`, we cannot ensure that we have sent the actual rows (they are incremented even while sending metadata).

Reply:
Thanks for checking. It does look a bit complicated. Lets go with sql_command way for now